src/CFamily/CMake/VsDevCmdEnvironmentProvider.cs

```diff
@@ -0,0 +1,231 @@
+                " && echo " + beginToken +                      // Output a marker so we known when to start processing the env vars
+                " && set" +                                     // Write out the env vars
+                " && echo " + endToken + " \"",                 // Output a marker to say we are done
+
+                CreateNoWindow = true,
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardInput = false,
+                RedirectStandardError = false,
+                StandardOutputEncoding = Encoding.Unicode,
+                StandardErrorEncoding = Encoding.Unicode
+            };
+
+            var capturedOutput = new List<string>();
+
+            var capturingOutput = false;
+
+            using (var process = processFactory.Start(startInfo))
+            {
+                process.HandleOutputDataReceived = data =>
+                {
+                    if (data == null) { return; }
+
+                    if (!capturingOutput && data.StartsWith(beginToken))
+                    {
+                        capturingOutput = true;
+                        return;
+                    }
+                    else if (capturingOutput && data.StartsWith(endToken))
+                    {
+                        capturingOutput = false;
+                        SafeKillProcess(process);
+                    }
+
+                    if (capturingOutput)
+                    {
+                        capturedOutput.Add(data);
+                    }
+                };
+
+                LogDebug($"Started process. Id: {process.Id}");
+                process.BeginOutputReadLine();
+
+                // Timeout in case something goes wrong.
+                process.WaitForExit(SCRIPT_TIMEOUT_MS);
+                if (process.HasExited)
+                {
+                    LogDebug("Process completed successfully");
+                }
+                else
+                {
+                    logger.WriteLine(Resources.VsDevCmd_TimedOut);
+                    capturedOutput = null;
+                    SafeKillProcess(process);
+                }
+            }
+
+            return capturedOutput;
+        }
+
+        private static Dictionary<string, string> ParseOutput(IList<string> capturedOutput)
+        {
+            if (capturedOutput == null)
+            {
+                return null;
+            }
+
+            var settings = new Dictionary<string, string>();
+
+            foreach (var item in capturedOutput)
+            {
+                var equalsIndex = item.IndexOf("=", StringComparison.OrdinalIgnoreCase);
+                if (equalsIndex > -1)
+                {
+                    var key = item.Substring(0, equalsIndex);
+                    var newValue = item.Substring(equalsIndex + 1);
+
+                    settings.Add(key, newValue);
```

Comment:
The VS version filters out environment variables that weren't changed by the script.
I don't think we need to do that - I'll check.